### ISSUE TITLE
Publish meilisearch-core and friends on crates.io?

### ISSUE THREAD

**Issue by trevyn** [meilisearch/MeiliSearch#952]
Curious if there's a reason this hasn't been done! It's certainly embeddable...

**Comment by MarinPostma** [comment]
Hello @trevyn !
This is certainly something we have in mind, and plan to do in a short future. In it's current form, Meilisearch core is not as embeddable as it may seem. It's interface have not been thought to be used without as a lib, and without http (and tokenizer, schema, etc...). 

That being said, we are currently working on a new engine, that will make meilisearch a lot faster, and plan to release it as a standalone library. Stay tuned :+1:

**Comment by trevyn** [comment]
Awesome, looking forward to it!!

**Comment by cryptoquick** [comment]
There would certainly be obstacles, but as apps with Rust-native desktop GUI libraries become more popular, this tool could become a natural go-to for document search capabilities in desktop apps as people start using Rust to replace clunky, large, and slow Electron apps, for example. I use Sled for this purpose currently.

**Comment by Kerollmops** [comment]
@cryptoquick thank you for saying that, this is really motivating to here that, fortunately we are working on making a real Rust library for the engine.

**Comment by maku** [comment]
@Kerollmops I really like MeiliSearch. So, I am curious...  Can you tell us how far the work on a lib has already progressed. Because I also think that it would be great to be able to use MeiliSearch in an app. I could think of to use it in a https://tauri.studio/en/ app.

**Comment by cryptoquick** [comment]
@maku yeah, similar, but in my case, it's for an Iced app: https://github.com/hecrj/iced

They recently released 0.2.0 and it's amazing.

**Comment by conanchen** [comment]
I am also trying to use millisearch in a https://tauri.studio/en/ app. 

the doc says milli core engine only handle one store and without update/multiple stores, it is not easy to embedded milli engine in a desktop applicaiton to take full function.  

Will it be nice to have an millisearch embeddable library to support belows?
- with managing multiple indexes, and handling the update store
- without actix_web

**Comment by Kerollmops** [comment]
Hey @conanchen,

I just created [an issue about what publishing Meilisearch and milli on crates.io](https://github.com/meilisearch/meilisearch/issues/2901) would bring us. Please comment on this specific issue if you have different concerns than ours.

> Will it be nice to have an millisearch embeddable library to support belows?
> 
>     * with managing multiple indexes, and handling the update store
>     * without actix_web

We are [currently rewriting the index scheduler](https://github.com/meilisearch/meilisearch/pull/2763), which is the main system that manages the tasks and operations that are executed on a set of indexes. You'll be able to use it on your own once it is merged on main and officially published. Internally, our HTTP server uses the index scheduler to operate on the indexes nothing more. You'll just have to extract the code from the routes.
